audit/hashstructure.go

```diff
@@ -108,18 +107,18 @@ func Hash(salter *salt.Salt, raw interface{}, nonHMACDataKeys []string) error {
 // the structure. Only _values_ are hashed: keys of objects are not.
 //
 // For the HashCallback, see the built-in HashCallbacks below.
-func HashStructure(s interface{}, cb HashCallback, ignoredKeys []string) (interface{}, error) {
-	s, err := copystructure.Copy(s)
+func HashStructure(s map[string]interface{}, cb HashCallback, ignoredKeys []string) (interface{}, error) {
```

Comment:
Please revert this change. The fact that the only things we pass in right now are map[string]interface{} doesn't change the fact that neither copystructure nor reflectwalk are able to work just fine on structs, and we may well want to pass in full structs at some future point.

Reply:
copystructure and reflectwalk work fine on structs, yes.  That doesn't hold true for hashWalker.

Reply:
Currently. It doesn't really bother me either way, it can be changed back, just seems like an unnecessary change.

Reply:
Yes, totally unnecessary.  I felt that without this I'd have to do more work to fill in the gaps, but if you prefer I can leave the signature alone and add a big disclaimer instead.

Reply:
Either way. Nobody is expecting you to code in support for non-maps. Keep this or leave a comment.